tests/unittest/_torch/thop/parallel/test_fp4_linear.py

```diff
@@ -76,3 +76,196 @@ def test_fp4_linear(dtype):
     # compare
     torch.cuda.synchronize()
     torch.testing.assert_close(output, output_ref)
+
+
+def pad_up(x, pad_size):
+    return (x + pad_size - 1) // pad_size * pad_size
+
+
+@skip_pre_blackwell
+@pytest.mark.parametrize("dtype", [torch.bfloat16])
+@pytest.mark.parametrize("mnk", [(128, 7168, 16384), (128, 24576, 1536),
+                                 (128, 2112, 7168), (128, 4096, 7168),
```

Comment:
I think we shall tests some `unalignd` m/n/k. 

If the Linear needs to handle generic shapes.